content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md

```diff
@@ -0,0 +1,300 @@
+To be able to do that, we will use a `LabelEncoder`. We have two categories of diagnosis status, `M` and `B`, the label encoder will change them to  categories of `1` and `0` respectively:
+```python
+lb = LabelEncoder()
+y = lb.fit_transform(df.diagnosis)
+```
+If you check `y` now, you will see that it transformed into an array of `1's` and `0's`:
+```python
+print(y)
+```
+![print(y)](/engineering-education/effects-of-standard-scaling-on-a-machine-learning-model/print-y.JPG)
+
+After that we can now go ahead and split our dataset into the training set and testing set in the ratios of 30:70:
+```python
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=1)
+```
+#### (iv) Fitting the model
+We will use `LinearSVC` to build a Support Vector Classifier model that categorizes our data into either one of the two diagnosis status:
+```python
+classifier = LinearSVC()
+# fitting model
+classifier.fit(X_train,y_train)
+# predict
+y_predict = classifier.predict(X_test)
+# check accuracy
+accuracy_b4 = classifier.score(X_test, y_test)
+```
+#### (v) Predicting
+We can then print both the prediction and testing sample and try to compare the results:
+```python
+print(y_predict)
+```
+![print(y_predict)](/engineering-education/effects-of-standard-scaling-on-a-machine-learning-model/y-predict.JPG)
+```python
+print(y_test)
+```
+![print(y_test)](/engineering-education/effects-of-standard-scaling-on-a-machine-learning-model/y-test.JPG)
+
+###  2.Using Standard Scaling
+We will build another model that uses `standard scaling` to level the dataset before fitting.
+####  (i) Creating and analyzing dataframe
+Use the code below to import the dataset and read then create the second dataframe:
+```python
+df1 = pd.read_csv(url,index_col=0)
+# dataframe summary
+df1.head(10)
+```
+Take note of how the values across the columns are distributed in a wide range. ie some are 122.80 and ,others  0.11840 :
+
+![df.head(10)](/engineering-education/effects-of-standard-scaling-on-a-machine-learning-model/df-head.JPG)
+####  (ii) Implementing standard scaling
+In order to scale properly, we will first collect the names of all the numerical columns in a list then use the list to fit  the standard scaler.
+```python 
+col =[]
+for col_name in df1.columns:
+    if df1[col_name].dtype=='object':
+        pass
+    else:
+        col.append(col_name)
+# fiting the standard scaler
+for i in col:
+    sc = StandardScaler().fit(df1[[i]])
+    df1[i] = sc.transform(df1[[i]]) 
+``` 
+Using `standard scaler` from `sklearn`, we were able to rescale our features using the `standardization` technique.
+Now if we check our `df1` again :
+```python
+df1.head(10)
+```
+ you will notice how the range between the values has been reduced:
```

Comment:
No need to separate the sentence that way. I've shown you an example above. Fix them where necessary.